ibis/expr/operations.py

```diff
@@ -696,6 +708,14 @@ class Mean(Reduction):
     output_type = rules.scalar_output(_mean_output_type)
 
 
+class Quantile(Reduction):
+
+    input_type = [value,
+                  number(name='quantile', allow_boolean=False),
+                  string(name='interpolation', optional=True)]
+    output_type = rules.scalar_output(_array_reduced_type)
```

Comment:
I don't remember how to spell the correct type for the input though, which is "any scalar or array of scalars".